net: conn_mgr: sntp: logging: backend: net: extend usage

maass-hamburg opened the pull request:
log backend net:
- When used when used with CONFIG_NET_IPV4_ACD and CONFIG_LOG_BACKEND_NET_USE_DHCPV4_OPTION a lot of warnings about dropped packages are generated, this fixes it. When used with the connection manager it is completely fixed, without it just at startup.
- can now use the connection manager to enable and disable the backend

net config init sntp:
- add option to use the connection manager

connection manager:
- set to unstable as it is used by multiple parts in zephyr and numerous samples

Fixes: #89643

maass-hamburg committed:
logging: backend: net: use connection mgr

Use connection manager to deactivate and activate
net log backend.
This removes the warnings about dropped packages
completly.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
net: config: init: sntp: use connection manager

Add option to use the connection manager to
schedule a (re-) sync on connection and diabeling the
work, when there is no connection.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
net: conn_mgr: set to unstable

Set connection manager to unstable. It is now used
by multiple parts in zephyr and numerous samples.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

jukkar commented on subsys/net/lib/config/init.c:
```diff
@@ -553,17 +553,8 @@ int net_config_init_app(const struct device *dev, const char *app_info)
-		const struct log_backend *backend = log_backend_net_get();
-
-		if (!log_backend_is_active(backend)) {
-			if (backend->api->init != NULL) {
-				backend->api->init(backend);
-			}
```
I think the idea here was that if the logging is not yet initialized, we need to do it here.

maass-hamburg replied:
this is part of the https://github.com/zephyrproject-rtos/zephyr/pull/89275. When CONFIG_LOG_BACKEND_NET_AUTOSTART is enabled, than the log backend net has the autostart flag set and will be automatically init by the logging subsystem

jukkar commented on subsys/logging/backends/log_backend_net.c:
```diff
@@ -338,9 +338,18 @@ static void panic(struct log_backend const *const backend)
 	panic_mode = true;
```
Typo in commit `logging: backend: net: avoid automatic enabling`
s/fixis/fixes/
s/than/then/
Also please start sentences with capital letters

maass-hamburg replied:
done

jukkar commented on subsys/net/lib/config/init.c:
```diff
@@ -553,17 +553,8 @@ int net_config_init_app(const struct device *dev, const char *app_info)
 	/* This is activated late as it requires the network stack to be up
```
Typo in commit `net: config: init: remove second init of syslog`
s/durring/during/

jukkar commented on subsys/net/lib/dhcpv4/dhcpv4.c:
```diff
@@ -1200,9 +1200,10 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
+			if (IS_ENABLED(CONFIG_LOG_BACKEND_NET_AUTOSTART) &&
+			    !IS_ENABLED(CONFIG_NET_CONFIG_SETTINGS)) {
+				log_backend_net_start();
+			}
```
Not sure how this is related to "avoiding automatic enabling"
Is there now two separate changes in this commit, one for automatic enabling and one for ACD?

maass-hamburg replied:
split the commit in two.

jukkar commented on include/zephyr/net/conn_mgr_connectivity.h:
```diff
@@ -26,7 +26,7 @@ extern "C" {
  * @brief Connection Manager Connectivity API
```
For the commit `net: conn_mgr: set to unstable`, please add information to commit msg why this change is needed.

maass-hamburg replied:
done

maass-hamburg committed:
logging: backend: net: avoid automatic enabling

Avoid automatic enabling of the syslog backend by
the logging subsystem.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
logging: backend: net: avoid early enabling

Avoid early enabling of the syslog backend in
the the dhcpv4 options parser.

When CONFIG_NET_IPV4_ACD is enabled, the assigned ip address has
not been checked, when the other dhcpv4 options are parsed, this would
lead to the syslog backend being enabled before the src ip address
is valid, so we get lots of warnings about dropd packets, this fixes
it at least on start. We will still get the warnings, when the iface
goes down and then up later.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg commented:
ping @jukkar
